streams/streams.md

```diff
@@ -0,0 +1,126 @@
+      validFiles.push(file);
+
+      cb();
+    });
+  }, () => {
+    console.log(validFiles);  // => Prints a list of all valid files
+  });
+});
+```
+
+We process each file in the `incrementals` folder in order by using [Async](https://github.com/caolan/async). Async is a module that helps you to deal with asynchronous challenges within Node. The `async.eachSeries` function we use here makes sure a function is applied to each item of the provided collection, but it only runs a single operation at a time. This makes sure all files are validated in order. This is necessary because we can have two sequential inremental files within the `incrementals` folder.
+
+Each file is split in lines. The first line should be `-- Increment timestamp: 20160129_192339`, the second line should be `-- Previous timestamp: 20160128_192500`. By splitting each line on space and only using the last element of each line, we have the timestamps and we can compare them to other timestamps available.
+
+So far, so good...
+
+## Drawbacks of this approach
+Though the code above works out fine for small files, it will result in errors for larger incremental files. Since we had to handle incremental files of about 200MB big, we soon ran into errors:
+
+```
+buffer.js:388
+    throw new Error('toString failed');
+    ^
+```
+
+This unclear error means the generated String takes up more memory than allowed (see [this issue for more information](https://github.com/nodejs/node/issues/3175)).
+
+When you think about this, it makes sense: when reading the file, the full contents of the file are stored in a buffer, which is cast to a string, which is turned into a large array of separate lines. That's a very memory-intensive process, only to retrieve the first two lines of a file...
+
+Let's find a better solution for this.
+
+## Better approach
+Node is asynchronous in handling I/O-bound tasks. Callbacks are often used for handling these asynchronous tasks, but Node has another way of handling this: [Streams](https://nodejs.org/api/stream.html). Streams come from Unix, where you might know them from using the `|` (pipe) in your shell. You are able to pipe data through multiple stages to its final destination, where each stage can transform the data. There are some excellent articles [written](https://github.com/substack/stream-handbook) [about](http://maxogden.com/node-streams.html) [Streams](https://www.sitepoint.com/basics-node-js-streams/) in Node.
+
+Node's `fs` module has a method `createReadStream`, to create a stream from a file. This stream does not buffer the entire contents of the file, but it emits an event as soon as a chunk of data is available. This is useful for us, since we'd only like to process the first two lines of a file and ignore the rest.
+
+The streams approach of our code would be:
+```js
+const fs = require('fs');
+const async = require('async');
+const path = require('path');
+const highland = require('highland');
+
+let latest = '20160128_192500';
```

Comment:
Maybe make it clear that this is not hardcoded, but the final state of a previous run.